chain/chain/src/store.rs

```diff
@@ -2680,10 +2695,17 @@ impl<'a> ChainStoreUpdate<'a> {
                 receipt,
             )?;
         }
-        for (hash, outcomes) in self.chain_store_cache_update.outcomes.iter() {
-            let mut existing_outcomes = self.chain_store.get_outcomes_by_id(hash)?;
-            existing_outcomes.extend_from_slice(outcomes);
-            store_update.set_ser(DBCol::TransactionResult, hash.as_ref(), &existing_outcomes)?;
+        for (outcome_id, outcomes) in self.chain_store_cache_update.outcomes.iter() {
```

Comment:
Should we instead change ChainStoreCacheUpdate::outcomes to be a map from (outcome_id, block_hash) to outcome?